android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingBannerFragment.java

```diff
@@ -0,0 +1,341 @@
+                ImageView githubIcon = mContentView.findViewById(R.id.link_github);
+                showAndSetListener(githubIcon, url);
+                break;
+
+            case REDDIT:
+                ImageView redditIcon = mContentView.findViewById(R.id.link_reddit);
+                showAndSetListener(redditIcon, url);
+                break;
+
+            case VIMEO:
+                ImageView vimeoIcon = mContentView.findViewById(R.id.link_vimeo);
+                showAndSetListener(vimeoIcon, url);
+                break;
+
+            case DISCORD:
+                ImageView discordIcon = mContentView.findViewById(R.id.link_discord);
+                showAndSetListener(discordIcon, url);
+                break;
+
+            case FACEBOOK:
+                ImageView facebookIcon = mContentView.findViewById(R.id.link_facebook);
+                showAndSetListener(facebookIcon, url);
+                break;
+
+            case INSTAGRAM:
+                ImageView instagramIcon = mContentView.findViewById(R.id.link_instagram);
+                showAndSetListener(instagramIcon, url);
+                break;
+        }
+    }
+
+    private void showAndSetListener(ImageView icon, String url) {
+        icon.setVisibility(View.VISIBLE);
+        icon.setTag(url);
+        icon.setOnClickListener(v -> {
+            String openUrl = (String) v.getTag();
+            openLink(openUrl);
+        });
+    }
+
+    private void openLink(String url) {
+        TabUtils.openUrlInNewTab(false, url);
+        dismissRewardsPanel();
+    }
+
+    private void setLogo() {
+        if (mBannerInfo == null) return;
+        String logo = mBannerInfo.getLogo();
+        if (!TextUtils.isEmpty(logo)) {
+            String prefix = IMAGE_URL_PREFIX;
+            if (logo.contains(prefix)) {
+                logo = logo.substring(prefix.length());
+                ImageView iv = (ImageView) mContentView.findViewById(R.id.publisher_favicon);
+                Glide.with(mActivity)
+                        .load(logo)
+                        .apply(RequestOptions.circleCropTransform())
+                        .into(iv);
+            }
+        }
+    }
+
+    private void setTitle() {
+        if (mBannerInfo == null) return;
+        TextView titleTextView = mContentView.findViewById(R.id.tipping_publisher_name);
+        String title = mBannerInfo.getName();
+        if (!TextUtils.isEmpty(title)) {
+            String provider = getProvider(mBannerInfo.getProvider());
+            if (!TextUtils.isEmpty(provider)) {
+                title = String.format(
+                        getResources().getString(R.string.title_on_provider), title, provider);
+            }
+            titleTextView.setText(title);
+        } else {
+            title = mBannerInfo.getTitle();
+            if (!TextUtils.isEmpty(title)) {
+                titleTextView.setText(title);
+            }
+        }
+    }
+
+    private String getProvider(String provider) {
```

Comment:
we can use `android:inputType="textCapWords"` for Textview instead of changing the text here.

Reply:
in that case, is it possible to use the same function as desktop through mojom ?

Reply:
It's not exposed , code is here
https://github.com/brave/brave-core/blob/master/components/brave_rewards/resources/shared/lib/publisher_platform.ts#L13